bcl2fastq/lib/bcl2fastq_utils.py

```diff
@@ -297,6 +297,15 @@ def construct_command(self):
         """
         raise NotImplementedError("Subclasses should implement this!")
 
+    def delete_output(self):
+        """
+        Delete the output directory if it exists
+        :return: None
+        """
+        if os.path.isdir(self.config.output):
```

Comment:
I agree that would make sense, but as it currently stands I have no ideas for exactly what to validate. I guess the same approach that I've used in other cases could be adopted, with one or more allowed output paths. Right now however, nothing like that is implemented.